### DISCUSSION TITLE
[Question][Hw2] Got some tests wrong when using Int32 library for arithmetic operations

### DISCUSSION THREAD

**Discussion by tncks0121** [prosyslab-classroom/cs524-program-analysis#292] [Q&A] [answered]
Name: 박수찬

YOUR QUESTION

I got 77/77 tests correct when I used `Int` to do integer arithmetic operations, while I got 76/77 tests correct when I used `Int32` instead.

In my repository, `master` branch got all tests correct, and `int32` branch got 76/77 tests correct. The difference between these branches is a single commit in the `int32` branch that does operations with `Int32` library.

I'm wondering if there is a test causing int32 overflow in the hidden tests. Also, given that signed integer overflow in C is undefined behavior, should I raise `Invalid_argument` if integer overflow happens?

**Comment by goodtaeeun** [comment, marked as answer]
Hi, you must indeed raise Invalid_argument for any undefined behaviors.
However, the specific test case you mentioned was a mistake, and we will fix it so that both of your branches can pass.

**Comment by tncks0121** [comment]
Checked that it doesn't affect the result, thanks.
